src/library/scala/concurrent/impl/Promise.scala

```diff
@@ -7,400 +7,441 @@
-   *  Twitter promise implementation]].
-   *
-   * In practice, flattening the chain cannot always be done perfectly. When a
-   * promise is added to the end of the chain, it scans the chain and links
-   * directly to the root promise. This prevents the chain from growing forwards
-   * But the root promise for a chain can change, causing the chain to grow
-   * backwards, and leaving all previously-linked promise pointing at a promise
-   * which is no longer the root promise.
-   *
-   * To mitigate the problem of the root promise changing, whenever a promise's
-   * methods are called, and it needs a reference to its root promise it calls
-   * the `compressedRoot()` method. This method re-scans the promise chain to
-   * get the root promise, and also compresses its links so that it links
-   * directly to whatever the current root promise is. This ensures that the
-   * chain is flattened whenever `compressedRoot()` is called. And since
-   * `compressedRoot()` is called at every possible opportunity (when getting a
-   * promise's value, when adding an onComplete handler, etc), this will happen
-   * frequently. Unfortunately, even this eager relinking doesn't absolutely
-   * guarantee that the chain will be flattened and that leaks cannot occur.
-   * However eager relinking does greatly reduce the chance that leaks will
-   * occur.
-   *
-   * Future.flatMap links DefaultPromises together by calling the `linkRootOf`
-   * method. This is the only externally visible interface to linked
-   * DefaultPromises, and `linkedRootOf` is currently only designed to be called
-   * by Future.flatMap.
-   */
-  // Left non-final to enable addition of extra fields by Java/Scala converters
-  // in scala-java8-compat.
-  class DefaultPromise[T] extends AtomicReference[AnyRef](Nil) with Promise[T] {
-
-    /** Get the root promise for this promise, compressing the link chain to that
-     *  promise if necessary.
-     *
-     *  For promises that are not linked, the result of calling
-     *  `compressedRoot()` will the promise itself. However for linked promises,
-     *  this method will traverse each link until it locates the root promise at
-     *  the base of the link chain.
-     *
-     *  As a side effect of calling this method, the link from this promise back
-     *  to the root promise will be updated ("compressed") to point directly to
-     *  the root promise. This allows intermediate promises in the link chain to
-     *  be garbage collected. Also, subsequent calls to this method should be
-     *  faster as the link chain will be shorter.
-     */
-    private def compressedRoot(): DefaultPromise[T] =
-      get() match {
-        case linked: DefaultPromise[_] => compressedRoot(linked)
-        case _ => this
-      }
+    override final def filter(@deprecatedName('pred) p: T => Boolean)(implicit executor: ExecutionContext): Future[T] = {
```

Comment:
Do we need the `@deprecatedName`?

Reply:
My suspicion is that you forked before those were removed in #6319

Reply:
@NthPortal Great catch, I thought I had found them all. I'll fix.